README.md

```diff
@@ -66,9 +61,24 @@ Finally, it provides this custom PHPStan type to assist static analysis:
 
 Notes:
 
+- **The underlying resource is not exposed publicly here.** The underlying resource MAY remain private or protected. See the _ResourceStream_ interface below for details on making underlying resource publicly accessible.
+
 - **There are no `isReadable()`, etc. methods.** If necessary, such functionality can be determined by typehinting against the interface, or by checking `instanceof`, etc.
 
-- **The `$metadata` property is expected change dynamically.** That is, as the resource gets read from and written to, the metadata for that resource is likely to change. Thus, the `$metadata` property value is expected to change along with it. In practical terms, this likely means a `stream_get_meta_data()` call on each access of `$metadata`.
+- **The `$metadata` property is expected change dynamically.** That is, as the underlying resource gets read from and written to, the metadata for that resource is likely to change. Thus, the `$metadata` property value is expected to change along with it. In practical terms, this likely means a `stream_get_meta_data()` call on each access of `$metadata`.
+
+### _ResourceStream_
+
+The _ResourceStream_ interface defines this property to allow public access to the underlying resource:
+
+- `public resource $resource { get; }`
+    - Represents the resource as if opened by [`fopen()`][].
+    - It MUST be a `resource of type (stream)`; for example, as determined by `get_resource_type()`.
+    - It SHOULD NOT be publicly settable.
+
+Notes:
+
+- **Not all _Stream_ implementations need to expose the underlying resource.** Exposing the resource gives full control over it to consumers, who can then manipulate it however they like (e.g. close it, move the pointer, and so on). However, having access to the resource may be necessary for some consumers.
```

Comment:
Note that not all streams might even have an underlying resource, a stream can be using an internal buffer, or using an FFI pointer, or anything else really.

Reply:
@azjezz Will do. Do you have examples of projects using an FFI pointer?

Reply:
@azjezz I have added language about non-resource-backed streams in [a separate commit](https://github.com/stream-interop/interface/commit/98bcdba470bf0d772e97c1141dd90a89f6db2ccb); please send a PR if you have suggestions for improvement.